library/src/handle.cpp

```diff
@@ -1,28 +1,78 @@
-
+/* ************************************************************************
+ * Copyright 2016 Advanced Micro Devices, Inc.
+ * ************************************************************************ */
 #include "definitions.h"
 #include "status.h"
 #include "handle.h"
 #include <hip/hip_runtime_api.h>
+#include <unistd.h>
+#include <pwd.h>
 
 /*******************************************************************************
  * constructor
  ******************************************************************************/
-_rocblas_handle::_rocblas_handle()
+_rocblas_handle::_rocblas_handle() : layer_mode(rocblas_layer_mode_logging)
 {
-
     // default device is active device
     THROW_IF_HIP_ERROR(hipGetDevice(&device));
     THROW_IF_HIP_ERROR(hipGetDeviceProperties(&device_properties, device));
 
     // rocblas by default take the system default stream 0 users cannot create
+
+    // set layer_mode from vaule of environment variable ROCBLAS_LAYER
+    char* str_layer_mode;
+    if((str_layer_mode = getenv("ROCBLAS_LAYER")) == NULL)
+    {
+        layer_mode = rocblas_layer_mode_none;
+    }
+    else
+    {
+        int int_layer_mode = atoi(str_layer_mode);
```

Comment:
You can read straight from atoi() into a rocblas_layer_mode type.  That could get rid of the temp variable.

Reply:
fixed